ethereum_history_api/circuits/lib/src/verifiers/receipt_test.nr

```diff
@@ -1,43 +1,65 @@
-use crate::verifiers::receipt::assert_tx_idx_equals;
+use dep::proof::utils::memcpy;
+use crate::fixtures::mainnet::{
+    cancun::small_block::{receipt::{encoded_receipt, receipt, tx_type}, header::number as block_number},
+    homestead::fork::{
+    receipt::{encoded_receipt as pre_byzantium_encoded_receipt, tx_type as legacy_tx_type},
+    header::number as pre_byzantium_block_number
+}
+};
+use crate::misc::arrays::{alter_array, sub_array_equals_up_to_length};
+use crate::verifiers::receipt::{
+    BYZANTIUM_BLOCK_NUM, assert_receipt_equals, split_receipt_into_tx_type_and_receipt_rlp,
+    assert_receipt_rlp_equal, MAX_RECEIPT_RLP_LENGTH, MAX_ENCODED_RECEIPT_LENGTH
+};
 
 #[test]
-fn test_assert_tx_idx_equals_zero() {
-    let tx_idx = 0x0;
-    let key = [0x0, 0x0, 0x0, 0x0, 0x8, 0x0];
-    assert_tx_idx_equals(key, tx_idx);
+fn test_split_receipt_into_tx_type_and_receipt_rlp() {
+    let (tx_type, receipt_rlp) = split_receipt_into_tx_type_and_receipt_rlp(tx_type, encoded_receipt);
+    assert(tx_type == encoded_receipt[0], "Incorrect tx type returned");
+    assert(receipt_rlp.len() == encoded_receipt.len() - 1, "Incorrect receipt RLP length returned");
+    assert(receipt_rlp[0] == encoded_receipt[1], "Incorrect receipt RLP returned");
```

Comment:
Why not receipt_rlp == encoded_receipt?

Reply:
Because encoded_receipt (in case of non legacy transactions type) has the first byte taken by tx_type, while rlp doesn't.

Reply:
Should we use `assert_subarray_equals`?